### PR TITLE
fix: vue 3 types, beta suffix & component name

### LINKED ISSUE #17480: TypeScript error when using props mounting option in `@cypress/vue`
### Current behavior

When type checking the following code:
```js
import { mount } from '@cypress/vue'
import HelloWorld from '../HelloWorld.vue'

mount(HelloWorld, { props: { msg: 'Hello Cypress' } })
```

TypeScript throws the following error:

```
src/components/__tests__/HelloWorld.spec.ts:6:25 - error TS2322: Type '{ msg: string; }' is not assignable to type 'VNodeProps & { __v_isVNode?: undefined; [Symbol.iterator]?: undefined; } & Record<string, any> & { $: ComponentInternalInstance; $data: {}; ... 11 more ...; $watch(source: string | Function, cb: Function, options?: WatchOptions<...> | undefined): WatchStopHandle; } & ... 4 more ... & ComponentCustomProperties'.
  Type '{ msg: string; }' is missing the following properties from type '{ $: ComponentInternalInstance; $data: {}; $props: Partial<{}> & Omit<Readonly<{ msg?: unknown; } & { msg: String; } & {}> & VNodeProps & AllowedComponentProps & ComponentCustomProps, never>; ... 10 more ...; $watch(source: string | Function, cb: Function, options?: WatchOptions<...> | undefined): WatchStopHandle; }': $, $data, $props, $attrs, and 10 more.

6     mount(HelloWorld, { props: { msg: 'Hello Cypress' } })
                          ~~~~~
```

### Desired behavior

No error is thrown, works as the `mount` method in `@vue/test-utils`.

### Test code to reproduce

https://github.com/sodatea/cypress-type-error

```sh
git clone https://github.com/sodatea/cypress-type-error.git
cd cypress-type-error
pnpm i
pnpm typecheck
```

### Cypress Version

8.0.0

### Other

Using `@cypress/vue` 3.0.0-beta.3 as it's the latest release.

Though, there's a `@cypress/vue` 3.0.1 released previously on npm. I have to pin the version to use beta.3. This should be fixed too.

### PR DESCRIPTION
**Pull request by elevatebart**
- Closes #17480

#### Main:
- Updated the types so that when using the `mount()` function. Users are now able to typecheck their tests suing vue-tsc.
- In the command log, display the name of the component with the mount function when we can guess it easily.

#### Additional
- Remove the "beta" suffix for 3.0 version of `@cypress/vue` to make sure it's installed normally (3.0.1 is already public)
- Add a typecheck in CI
- Add some typescript vue tests since typescript is a feature of vue 3
- Add an example of `<script setup>` component and its test

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue or this PR been tagged with a release in ZenHub? 
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: npm/vue/src/index.ts [hunk 1]
```diff
@@ -53,17 +58,162 @@ Cypress.on('run:start', () => {
...
   })
 })
 
-export function mount<Props = any> (
-  comp: Component<Props>,
-  options: CyMountOptions<Props> = {},
+type PublicProps = VNodeProps & AllowedComponentProps & ComponentCustomProps;
+
+// Class component - no props
+export function mount<V>(
```
Comment (lmiller1990) [inline npm/vue/src/index.ts:73]: Can we put a link to where these came from? So if someone looks at in the future and wants to update they can copy paste again - actually reading this type is basically impossible unless you know Vue 3 inside-out

File: npm/vue/src/index.ts [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
+  Component, DefineComponent, FunctionalComponent, ComputedOptions,
+  MethodOptions, ComponentOptionsMixin, EmitsOptions, ComponentOptionsWithObjectProps, ComponentPropsOptions, ComponentOptionsWithArrayProps, ComponentOptionsWithoutProps } from 'vue'
 import { MountingOptions, VueWrapper, mount as VTUmount } from '@vue/test-utils'
 import {
   injectStylesBeforeElement,
   StyleOptions,
   ROOT_ID,
   setupHooks,
 } from '@cypress/mount-utils'
+import { GlobalMountOptions } from '@vue/test-utils/dist/types'
```
Comment (lmiller1990) [inline npm/vue/src/index.ts:14]: Fine for now but we should export this from the root of VTU

File: npm/vue/src/index.ts [hunk 3]
```diff
@@ -119,17 +273,34 @@ export function mount<Props = any> (
...
+function getComponentDisplayName (componentOptions: any): string {
+  if (componentOptions.name) {
+    return componentOptions.name
+  }
+
+  if (componentOptions.__file) {
+    const filepathSplit = componentOptions.__file.split('/')
+    const fileName = filepathSplit[filepathSplit.length - 1]
+
+    return fileName.replace(/\....?$/, '')
```
Comment (lmiller1990) [inline npm/vue/src/index.ts:285]: What does this do? Can you put a comment explaning (very hard to understand regexp without knowing expected input)

File: npm/vue/cypress/support/index.js [hunk 4]
```diff
@@ -1,3 +1,3 @@
 // created automatically when cy:open starts
-
+import 'regenerator-runtime/runtime'
```
Comment (lmiller1990) [inline npm/vue/cypress/support/index.js:2]: Why do we need this? Can we just update tsconfig to target `esnext`

**Review by lmiller1990 — COMMENTED** [review]
Left some comments

File: npm/vue/src/index.ts (hunk 3 above)
Comment (elevatebart) [reply npm/vue/src/index.ts:285]: yep will do

File: npm/vue/cypress/support/index.js (hunk 4 above)
Comment (elevatebart) [reply npm/vue/cypress/support/index.js:2]: You are 💯 right we should get rid of babel in there...

Comment (elevatebart) [reply npm/vue/cypress/support/index.js:2]: It's just that... webpack 😇

File: npm/vue/src/index.ts (hunk 1 above)
Comment (elevatebart) [reply npm/vue/src/index.ts:81]: Done !!

File: npm/vue/src/index.ts (hunk 2 above)
Comment (elevatebart) [reply npm/vue/src/index.ts]: I found another way ! No need to export anything to keep the types tight.

File: npm/vue/src/index.ts (hunk 3 above)
Comment (elevatebart) [reply npm/vue/src/index.ts]: Done

File: npm/vue/cypress/support/index.js (hunk 4 above)
Comment (elevatebart) [reply npm/vue/cypress/support/index.js]: OK it's been removed now. since it was unnecessary.

**Review by lmiller1990 — APPROVED** [review]

**Pull request merged**